src/java/org/apache/cassandra/index/sai/utils/RangeIntersectionIterator.java

```diff
@@ -57,6 +54,80 @@ public static boolean shouldDefer(int numberOfExpressions)
         return (INTERSECTION_CLAUSE_LIMIT <= 0) || (numberOfExpressions <= INTERSECTION_CLAUSE_LIMIT);
     }
 
+    private final List<RangeIterator> ranges;
+    private final List<RangeIterator> toRelease;
+    private final List<RangeIterator> processedRanges;
+
+    private RangeIntersectionIterator(Builder.Statistics statistics, List<RangeIterator> ranges)
+    {
+        super(statistics);
+        this.ranges = ranges;
+        this.toRelease = new ArrayList<>(ranges);
+        this.processedRanges = new ArrayList<>(ranges.size());
+    }
+
+    protected PrimaryKey computeNext()
```

Comment:
I remember I proposed a much simpler, equivalent code for this. What happened to that?